### DISCUSSION TITLE
How can I set the data manually?

### DISCUSSION THREAD

**Discussion by NicolasSacC** [filamentphp/filament#4719] [Help]
Hi!

I am working on an edit form for my User model. However, I need to use a custom script that will take the data and set it in the model by Javascript.

I created a custom "EditUser" page to put the script.

When the option in Google Autocomplete is selected, it sets the value on the Livewire component:
```
<script src="https://maps.googleapis.com/maps/api/js?key={{ config('api.maps') }}&libraries=places"></script>

<script type="text/javascript">
    window.onload = () => {
        const input = document.getElementById("full_address");
        const autocomplete = new google.maps.places.Autocomplete(input, {
            fields: ["place_id", "geometry", "formatted_address", "name"],
        });

        autocomplete.addListener("place_changed", () => {
            const place = autocomplete.getPlace();

            console.log(place);

            document.getElementById('full_address').value = place.formatted_address;
            @this.set('full_address', place.formatted_address)

            document.getElementById('place_id').value = place.place_id;
            @this.set('place_id', place.place_id)

            if (!place.geometry || !place.geometry.location) {
                return;
            }
        });
    }
```

It works on my other Filament forms. But in this case (in the admin panel), I get a `Property [$full_address] not found on component: [app.filament.resources.user-resource.pages.edit-user]`.

Where are the model properties stored in the class? How can I set the data from my JS like I would do in any other Filament form/Livewire component?

**Comment by stephenlake** [comment]
Bit of a necro, but needed to figure this out too. This question put me on the right track and encountered the same error.

You can use `$getStatePath()` to get the full property path.  `@this.set('{{ "$getStatePath().'.full_address' }}', 'whatever');`
